### PR TITLE
SONARJS-609 Rule: Function parameters with default values should be last

### PR DESCRIPTION
**Pull request by vilchik-elena**

### CODE REVIEW

File: javascript-checks/src/main/java/org/sonar/javascript/checks/DefaultParametersNotLastCheck.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+import org.sonar.plugins.javascript.api.visitors.SubscriptionVisitorCheck;
+import org.sonar.squidbridge.annotations.ActivatedByDefault;
+import org.sonar.squidbridge.annotations.SqaleConstantRemediation;
+import org.sonar.squidbridge.annotations.SqaleSubCharacteristic;
+
+@Rule(
+  key = "S1788",
+  name = "Function parameters with default values should be last",
+  priority = Priority.CRITICAL,
+  tags = {Tags.ES2015, Tags.BUG, Tags.PSR2})
```
Comment (pynicolas) [inline javascript-checks/src/main/java/org/sonar/javascript/checks/DefaultParametersNotLastCheck.java]: @vilchik-elena I don't think we should set the PSR2 tag: this is specific to PHP.

File: javascript-checks/src/test/resources/checks/DefaultParametersNotLast.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+}
+
+foo((p1 = 1, p2) => {   // Noncompliant [[sc=6;ec=8]]
+});
+
+function foo({a, b} = {a:42, b:"hello"}, p1){ // Noncompliant {{Move parameter "{a, b}" after parameters without default value.}}
+}
+
+function foo(p1 = 1, ...restArgs) {  // Noncompliant
+}
```
Comment (pynicolas) [inline javascript-checks/src/test/resources/checks/DefaultParametersNotLast.js]: @vilchik-elena I would add at least one compliant case.

Comment (vilchik-elena) [inline javascript-checks/src/test/resources/checks/DefaultParametersNotLast.js]: @pynicolas indeed, thanks!

File: javascript-checks/src/main/java/org/sonar/javascript/checks/DefaultParametersNotLastCheck.java (hunk 1 above)
Comment (vilchik-elena) [inline javascript-checks/src/main/java/org/sonar/javascript/checks/DefaultParametersNotLastCheck.java]: @pynicolas Agree. BTW WDYT about "misra"?

Comment (pynicolas) [inline javascript-checks/src/main/java/org/sonar/javascript/checks/DefaultParametersNotLastCheck.java]: @vilchik-elena Currently, we have 17 rules tagged with "misra".

Comment (vilchik-elena) [inline javascript-checks/src/main/java/org/sonar/javascript/checks/DefaultParametersNotLastCheck.java]: @pynicolas And what does this statistics mean?:)

Comment (pynicolas) [inline javascript-checks/src/main/java/org/sonar/javascript/checks/DefaultParametersNotLastCheck.java]: @vilchik-elena We have rules tagged with "misra", that's all. However, I don't want to add "psr2". What did you mean with your initial question: "BTW WDYT about "misra"?"?

**Pull request merged**